### ISSUE TITLE
[a11y] 11.1.4.11 Non-text contrast

### ISSUE THREAD

**Issue by tschiebel** [owncloud/android#4365]
audit: https://infinite.owncloud.com/s/QtwuiGHikwFjozV

https://www.etsi.org/deliver/etsi_en/301500_301599/301549/03.02.01_60/en_301549v030201p.pdf
https://github.com/BIK-BITV/BIK-App-Test/tree/main/Pr%C3%BCfschritte/de

Note: Information-bearing graphics and graphic control elements as well as their states should have a contrast of at least 3:1 to adjacent colors.

### Personal Space

- [x] The edges of the "Search folder" input field and the button for deleting search entries do not meet the contrast requirements for graphical control elements.
![image](https://github.com/owncloud/android/assets/132444655/002e3a55-6d2c-47fe-87ae-8a0b1448951a)

### Edit shared Link

- [x] Adjust contrast for switches (if necessary)

Note: If the two switches are not native elements, the contrasts should be adjusted accordingly. Only native switches
are an exception.

### Spaces

- [x] "Search folder" input field and the button for deleting search entries - See comment on Personal Space.

**Comment by jesmrec** [comment]
- "Search folder" text field does not have visible edges on purpose. Is that right?

**Comment by jesmrec** [comment]
It will move to an approach like https://m3.material.io/components/search/overview, with a bubble-like. Needs a first research to assure it is feasible.

**Comment by JuancaG05** [comment]
We'll address here the changes related to the toolbar mentioned in https://github.com/owncloud/android/issues/4364

**Comment by JuancaG05** [comment]
> - Adjust contrast for switches (if necessary)
>
> Note: If the two switches are not native elements, the contrasts should be adjusted accordingly. Only native switches
are an exception.

This one won't be addressed because switches are native elements

**Comment by Aitorbp** [comment]
Research:
First of all, we need to know that there are two toolbars currently working in the application: `RootToolbar` y `StandardToolbar`. 

The `RootToolbar` is displayed in the following views: 
- `Root List File`
- `Spaces view`
- `Upload View`

The `Standard toolbar` appears in the following views: 
- Set target path for automatic uploads
- List files (not root)
- Receiving external files

In this [PR](https://github.com/owncloud/android/pull/4433) we are trying to unify the behaviour of the toolbar to the `onCreateOptionsMenu` method in `ToolbarActivity`. We are moving the general behaviour of the toolbar from the child classes to the `ToolbarActivity`, removing the useless code in the child classes. So everything related to colors is now handled in `ToolbarActivity` and the corner cases in its classes override the `onCreateOptionsMenu` function.

This way, the following classes don't need to display the main_menu, so we override this method to return `false` (won't be displayed):
- `UploadListActivity`
- `ShareActivity`
- `PreviewImageActivity`
- `PreviewVideoActivity`
We need to do this now because these classes extend from `ToolbarActivity` and `main_menu` is attached to it. So we have to do this behaviour.

Otherwise we have fixed a bug. Now when we are in the spaces view, the text hint of the SearchView is `Search spaces`.
